components/camel-arangodb/src/main/java/org/apache/camel/component/arangodb/ArangoDbEndpoint.java

```diff
@@ -0,0 +1,197 @@
+    private String host;
+    @UriParam
+    private int port;
+    @UriParam(label = "security", secret = true)
+    private String user;
+    @UriParam(label = "security", secret = true)
+    private String password;
+    @UriParam
+    private String collection;
+    @UriParam
+    private ArangoDbOperation operation;
+
+    public ArangoDbEndpoint() {
+    }
+
+    public ArangoDbEndpoint(String uri, ArangoDbComponent component) {
+        super(uri, component);
+    }
+
+    public Producer createProducer() {
+        return new ArangoDbProducer(this);
+    }
+
+    public Consumer createConsumer(Processor processor) {
+        throw new UnsupportedOperationException("You cannot receive messages at this endpoint: " + getEndpointUri());
+    }
+
+    public String getDatabase() {
+        return database;
+    }
+
+    /**
+     * database name
+     *
+     * @param database
+     */
+    public void setDatabase(String database) {
+        this.database = database;
+    }
+
+    public ArangoDB getArango() {
+        return arango;
+    }
+
+    public void setArango(ArangoDB arango) {
+        this.arango = arango;
+    }
+
+    public String getHost() {
+        return host;
+    }
+
+    /**
+     * host if host and/or port different from default
+     *
+     * @param host
+     */
+    public void setHost(String host) {
+        this.host = host;
+    }
+
+    public int getPort() {
+        return port;
+    }
+
+    /**
+     * port if  host and/or port different from default
+     *
+     * @param port
+     */
+    public void setPort(int port) {
+        this.port = port;
+    }
+
+    public String getUser() {
+        return user;
+    }
+
+    /**
+     * user if user and/or password different from default
+     *
+     * @param user
+     */
+    public void setUser(String user) {
+        this.user = user;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    /**
+     * password if user and/or password different from default
+     *
+     * @param password
+     */
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+
+    public String getCollection() {
+        return collection;
+    }
+
+    /**
+     * collection in the database
+     *
+     * @param collection
+     */
+    public void setCollection(String collection) {
+        this.collection = collection;
+    }
+
+    public ArangoDbOperation getOperation() {
+        return operation;
+    }
+
+    /**
+     * operation to perform
+     *
+     * @param operation
+     */
+    public void setOperation(ArangoDbOperation operation) {
+        this.operation = operation;
+    }
+
+    @Override
+    protected void doStart() throws Exception {
+        super.doStart();
+
+        if (arango == null) {
+
+            final ArangoDB.Builder builder = new ArangoDB.Builder();
+
+            if (host != null) {
+                builder.host(host, port);
+            }
+
+            if (user != null) {
```

Comment:
You can use the ObjectHelper methods for checking

Reply:
done